scripts/download_checkpoints.py

```diff
@@ -0,0 +1,137 @@
+   base_path.mkdir(parents=True, exist_ok=True)
+   print(f"Saving to: {base_path}")
+   available_files = try_get_directory_listing(public_url)
+   
+   if not available_files:
+       raise ValueError("No matching files found in directory")
+   
+   failed_files = []
+   for file in available_files:
+       file_url = urljoin(public_url.rstrip('/') + '/', file)
+       file_path = base_path / file
+       try:
+           print(f"\nDownloading: {file}")
+           download_file(file_url, file_path)
+       except requests.exceptions.Timeout:
+           print(f"Timeout error for {file}, retrying once...")
+           try:
+               download_file(file_url, file_path)
+           except requests.exceptions.RequestException as e:
+               failed_files.append(file)
+               print(f"Failed to download {file}: {e}")
+       except requests.exceptions.RequestException as e:
+           failed_files.append(file)
+           print(f"Failed to download {file}: {e}")
+   if failed_files:
+       print(f"\nWARNING: Failed to download these files: {failed_files}")
+
+def main():
+    parser = argparse.ArgumentParser(description='Download OLMo checkpoints')
+    subparsers = parser.add_subparsers(dest='command', help='Commands')
+    download_parser = subparsers.add_parser('download', 
+                                          help='Download checkpoints from CSV file')
+    download_parser.add_argument('csv_file', type=str, 
+                               help='Path to the CSV file containing checkpoint URLs')
+    download_parser.add_argument('--step', type=str, required=True,
+                               help='Specific step number to download')
+    download_parser.add_argument('--save-dir', type=str, default='./checkpoints',
+                               help='Base directory to save downloaded checkpoints')
+    list_parser = subparsers.add_parser('list',
+                                       help='List available checkpoint steps')
+    list_parser.add_argument('csv_file', type=str,
+                            help='Path to the CSV file containing checkpoint URLs')
+    args = parser.parse_args()
+    
+    print(f"Reading CSV file: {args.csv_file}")
+    
+    with open(args.csv_file, 'r') as f:
+        reader = csv.DictReader(f)
+        urls = [(row['Step'], row['Checkpoint Directory']) for row in reader]
+    
+    if args.command == 'list':
+        print("Available steps:")
+        for step, _ in urls:
+            print(f"Step {step}")
+        return
+    
+    if args.step:
+        urls = [(step, url) for step, url in urls if step == args.step]
+        if not urls:
+            print(f"Error: Step {args.step} not found in the CSV file.")
+            print("Use list argument to see available step numbers.")
+            return
+    
+    print(f"Saving checkpoints to: {args.save_dir}")
+    for step, url in urls:
+        r2_url = convert_to_r2_url(url)
+        public_url = convert_to_public_url(r2_url)
```

Comment:
The URLs in the CSV are already public URLs? Why are we doing this change?

Reply:
I've taken these functions from my another file where I will have r2 urls in csv while uploading. I missed this here.